src/compiler/scala/tools/nsc/typechecker/Checkable.scala

```diff
@@ -127,14 +127,20 @@ trait Checkable {
   private def uncheckedOk(tp: Type) = tp hasAnnotation UncheckedClass
 
   private def typeArgsInTopLevelType(tp: Type): List[Type] = {
-    val tps = tp match {
-      case RefinedType(parents, _)              => parents flatMap typeArgsInTopLevelType
-      case TypeRef(_, ArrayClass, arg :: Nil)   => if (arg.typeSymbol.isAbstractType) arg :: Nil else typeArgsInTopLevelType(arg)
-      case TypeRef(pre, sym, args)              => typeArgsInTopLevelType(pre) ++ args
-      case ExistentialType(tparams, underlying) => tparams.map(_.tpe) ++ typeArgsInTopLevelType(underlying)
-      case _                                    => Nil
+    val res: ListBuffer[Type] = ListBuffer.empty[Type]
+    def add(t: Type) = if (!isUnwarnableTypeArg(t)) res += t
+    def loop(tp: Type): Unit = tp match {
+      case RefinedType(parents, _)              => parents foreach loop
+      case TypeRef(_, ArrayClass, arg :: Nil)   => if (arg.typeSymbol.isAbstractType) add(arg) else loop(arg)
+      case TypeRef(pre, sym, args)              =>
+        loop(pre)
+        args.foreach(add)
+      case ExistentialType(tparams, underlying) =>
+        tparams.foreach(tp => add(tp.tpe))
+        loop(underlying)
+      case _                                    => ()
     }
-    tps filterNot isUnwarnableTypeArg
+    res.toList
```

Comment:
Just need to add `loop(tp)` here 😄

Reply:
I just `push -f`-ed that change.

Reply:
Ah, yes. I was wondering what could be failing. Thanks!
I am re-pushing with other style changes.